lib/getInstanceId.js

```diff
@@ -12,11 +12,31 @@ var FirebaseError = require("./error");
  */
 module.exports = function(options) {
   return api.getProject(options.project).then(function(project) {
-    if (!_.has(project, ["instances", "database", 0])) {
+    if (!_.has(project, ["instances", "database"])) {
       throw new FirebaseError("No instance found for project. Please try a different project.", {
         exit: 1,
       });
     }
-    return project.instances.database[0];
+    if (!_.isArray(project.instances.database)) {
+      throw new FirebaseError("No instance found for project. Please try a different project.", {
+        exit: 1,
+      });
+    }
+    if (!_.has(options, ["instance"])) {
+      return project.instances.database[0];
+    } else if (_.includes(project.instances.database, options.instance)) {
+      return options.instance;
+    } else {
+      throw new FirebaseError(
+        "No instance named '" +
+          options.instance +
+          "' found for project. Available options: [" +
+          project.instances.database.join(", ") +
+          "] Please try a different project.",
```

Comment:
Revise last sentence to: "Please choose a different instance on this project, or switch to a different project using `firebase use`"